cmake/Findo3de.cmake

```diff
@@ -21,24 +23,35 @@ o3de_current_file_path(current_path)
 # because later it's read again using a path like ${LY_ROOT_FOLDER}/engine.json, which
 # is also normalized.  They should match to avoid errors on some build systems.
 cmake_path(SET engine_json_path NORMALIZE ${current_path}/../engine.json)
-file(READ ${engine_json_path} engine_json)
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${engine_json_path})
 
-string(JSON this_engine_name ERROR_VARIABLE json_error GET ${engine_json} engine_name)
-if(json_error)
-    message(FATAL_ERROR "Unable to read key 'engine_name' from '${engine_json_path}', error: ${json_error}")
-endif()
+if(NOT LY_ENGINE_NAME_TO_USE)
```

Comment:
Can we change this name to `O3DE_ENGINE_NAME_TO_USE`?

Reply:
I left it in there for backwards compatibility but the new code no longer defines `O3DE_ENGINE_NAME_TO_USE`, rather `Version.cmake` defines `O3DE_ENGINE_NAME`.